pkg/fab/discovery/discovery.go

```diff
@@ -29,20 +28,27 @@ const (
 	signerCacheSize = 10 // TODO: set an appropriate value (and perhaps make configurable)
 )
 
+//Client gives ability to send discovery request to multiple targets.
+//There are cases when multiple targets requested and some of them are hanging, recommended to cancel ctx after first successful response.
+//Note: "access denied" is a success response, so check for it after response evaluation.
+type Client interface {
```

Comment:
I wouldn't define the interface here. In Go, interfaces are (usually) defined at the receiving end. The Client struct doesn't make use of this interface at all. If you wish to define it in a common place then it should be defined in pkg/client/common.

Reply:
I'm considering discovery service as a component(like net.Dialer), so coupling on this package is ok.

What if I make Client struct as a private one? no reason to expose it, no functionality beyond interface.